app/create-pr.js

```diff
@@ -0,0 +1,171 @@
+async function createPr(params, debug = false) {
+  /** auth with github */
+  const octokit = new Octokit({ auth });
+
+  /** get params */
+  const { branch, title, body, files = [] } = params || {};
+
+  /** check for missing params */
+  const missing = [];
+  if (!branch) missing.push("branch");
+  if (!title) missing.push("title");
+  if (!body) missing.push("body");
+  if (!files.length) missing.push("files");
+  if (missing.length) throw Error(`Insufficient ${missing.join(", ")}`);
+
+  /** get main branch */
+  let main;
+  try {
+    main = (
+      await octokit.rest.git.getRef({
+        owner,
+        repo,
+        ref: "heads/main",
+      })
+    ).data.object.sha;
+  } catch (error) {
+    if (debug) console.warn(error);
+    throw Error(`Couldn't get main branch: ${error?.response?.data?.message}`);
+  }
+
+  /** get existing branch */
+  let newBranch;
+  try {
+    newBranch = await octokit.rest.git.getRef({
+      owner,
+      repo,
+      ref: `heads/${branch}`,
+    });
+  } catch (error) {
+    if (debug) console.warn(error);
+    console.warn(
+      `Couldn't get existing branch ${branch}: ${error?.response?.data?.message}`,
+    );
+  }
+
+  if (!newBranch)
+    /** create new branch */
+    try {
+      newBranch = await octokit.rest.git.createRef({
+        owner,
+        repo,
+        ref: `refs/heads/${branch}`,
+        sha: main,
+      });
+    } catch (error) {
+      if (debug) console.warn(error);
+      throw Error(
+        `Couldn't create new branch ${branch}: ${error?.response?.data?.message}`,
+      );
+    }
+
+  /** update files */
+  for (const [index, { path, content }] of Object.entries(files)) {
+    if (!path || !content) throw Error(`File ${index} insufficient`);
+
+    /** get existing file */
+    let existing;
+    try {
+      existing = (
+        await octokit.rest.repos.getContent({
+          owner,
+          repo,
+          branch,
+          path,
+        })
+      ).data.sha;
+    } catch (error) {
+      if (debug) console.warn(error);
+      console.warn(
+        `Couldn't get existing file ${path}: ${error?.response?.data?.message}`,
+      );
+    }
+
+    /** create or update file */
+    try {
+      await octokit.rest.repos.createOrUpdateFileContents({
+        owner,
+        repo,
+        branch,
+        path,
+        message: title,
+        content: Buffer.from(
+          content.endsWith("\n") ? content : content + "\n",
+        ).toString("base64"),
+        sha: existing,
+      });
+    } catch (error) {
+      if (debug) console.warn(error);
+      throw Error(
+        `Couldn't create/update file ${path}: ${error?.response?.data?.message}`,
+      );
+    }
+  }
+
+  let pr;
+
+  /** create pull request */
+  try {
+    pr = await octokit.rest.pulls.create({
+      owner,
+      repo,
+      base: "main",
+      head: branch,
+      title,
+      body,
+    });
+  } catch (error) {
+    if (debug) console.warn(error);
+    throw Error(
```

Comment:
This possibly could be a non-critical error, since above we allow updating files on an existing branch... so theoretically that would just update any existing PR for that branch. However, we still need to return PR details here (to show the user a link to it), and apparently looking up a PR from a branch name is actually non-trivial because of some GitHub-specific weirdness. Maybe something to consider in the future.